Gem/Code/Source/Components/UI/UiSettingsComponent.cpp

```diff
@@ -140,91 +115,132 @@ namespace MultiplayerSample
     {
     }
 
+    template<typename ValueType>
+    uint32_t UiSettingsComponent::GetRotatedIndex(
+        const AZStd::vector<AZStd::pair<ValueType, AZStd::string>>& valuesToLabels,
+        const ValueType& value, ToggleDirection toggleDirection)
+    {
+        const size_t totalValues = valuesToLabels.size();
+
+        uint32_t curIndex = 0;
+
+        // Loop through and look for the correct value
+        for (size_t index = 0; index < totalValues; index++)
```

Comment:
nit: use a range loop instead ?

Reply:
You mean like `for (auto& value : valuesToLabels)`? It's messy to do that when you need the index value (see line 132), since you then need to track the index as a separate incrementing counter. IMO it makes it less readable that just the good old fashioned index loop.